routing/pathfind.go

```diff
@@ -346,10 +332,13 @@ func newRoute(amtToSend, feeLimit lnwire.MilliSatoshi, sourceVertex Vertex,
 			return nil, newErrf(ErrFeeLimitExceeded, err)
 		}
 
-		// As a sanity check, we ensure that the selected channel has
-		// enough capacity to forward the required amount which
-		// includes the fee dictated at each hop.
-		if nextHop.AmtToForward.ToSatoshis() > nextHop.Channel.Capacity {
+		// As a sanity check, we ensure that the incoming channel has
+		// enough capacity to carry the required amount which
+		// includes the fee dictated at each hop. Make the comparison
+		// in msat to prevent rounding errors.
+		if nextHop.AmtToForward + fee > lnwire.NewMSatFromSatoshis(
```

Comment:
This is actually the _outgoing_ channel though. Also at this point w/ the old code, the `amtToForward` already factors in the fee added at the prior hop. The fee is the difference between the incoming and outgoing amount to a target node.

Reply:
I _thiiink_ this is correct now.